### PR TITLE
Implement getGroupGames()

### PR DESCRIPTION
**Pull request by alanbixby**
**This is a duplicated fork of #312, but has a readable commit/file history, and has some logic errors fixed.**

~~**NOTE:** A test case was added, but currently it has to point to one of my test groups instead of `noblox.js` as some reason all new games I try to create become personal games instead of group games.. possibly a symptom of their experience/game rushed rebranding.~~ Test now uses the `noblox.js` test group.

Screenshot of the function working:

image

image

### CODE REVIEW

File: test/group.test.js [hunk 1]
```diff
@@ -111,6 +111,27 @@ describe('Group Methods', () => {
     })
   })
 
+  it('getGroupGames() returns an array of group games', () => {
+    return getGroupGames({ groupId: 9997719, limit: 1 }).then((res) => { // FIXME: Convert to noblox.js groupId, currently Roblox broke the creation of group games during the experience rebranding
```
Comment (suufi) [inline test/group.test.js:115]: Left a fix me.

File: typings/index.d.ts [hunk 2]
```diff
@@ -39,6 +39,17 @@ declare module "noblox.js" {
         IsPrimary: boolean,
     }
 
+    interface GroupGameInfo {
+        id: number;
+        name: string;
+        description: string | null;
```
Comment (suufi) [inline typings/index.d.ts:45]: `description?` maybe?

File: typings/index.d.ts [hunk 3]
```diff
@@ -39,6 +39,17 @@ declare module "noblox.js" {
...
+        creator: {id: number; type: string;};
```
Comment (suufi) [inline typings/index.d.ts:46]: tad messy, could do `{ id: number, type: string};`
same for L47

File: typings/jsDocs.ts [hunk 4]
```diff
@@ -774,6 +774,20 @@ type Group = {
     isLocked: boolean;
 }
 
+/**
+ * @typedef
+ */
+type GroupGameInfo = {
```
Comment (suufi) [inline typings/jsDocs.ts:780]: see index.d.ts changes

**Review by suufi — CHANGES_REQUESTED** [review]
Looks good for the most part, small things I've noticed.

File: typings/index.d.ts (hunk 2 above)
Comment (alanbixby) [reply typings/index.d.ts]: Fixed and I corrected the types for some other interfaces that used null unions instead of optional parameters.

**Review by suufi — APPROVED** [review]

**Comment by Vilksian** [comment]
I'll close my original pull request.

**Pull request merged**